web/scripts/dev/cli.ts

```diff
@@ -0,0 +1,215 @@
+import { performance } from "perf_hooks"
+import formatMS from "pretty-ms"
+import readline from "readline"
+import {
+  answerYesOrNo,
+  error,
+  info,
+  infoline,
+  linebreak,
+  pc,
+  processes,
+  question,
+  section,
+  separator,
+  warn
+} from "../pretty-logs"
+import { Containers } from "./containers"
+import { Mockoon } from "./mockoon"
+import { closing, isBuild, isClean, isServe, pnpm, starting } from "./util"
+import { Vite } from "./vite"
+
+// TODO: add shortcut for rebuilding a container, and then restarting it
+
+export class DevCLI {
+  declare vite?: Vite
+  declare mockoon?: Mockoon
+  declare containers?: Containers
+
+  stopped = false
+  starting = false
+
+  async clean() {
+    if (this.stopped) return
+    this.stopped = true
+
+    // try to stop anything that might still be running
+    for (const proc of processes) {
+      if (!proc.killed) {
```

Comment:
Does this happen after you've sent `SIGTERM` and already given it some time to try and clean up on its own?

Reply:
What do you mean?